langchain/chains/sql_database/base.py

```diff
@@ -74,34 +83,68 @@ def _call(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
         table_info = self.database.get_table_info(table_names=table_names_to_use)
         llm_inputs = {
             "input": input_text,
-            "top_k": self.top_k,
+            "top_k": str(self.top_k),
             "dialect": self.database.dialect,
             "table_info": table_info,
             "stop": ["\nSQLResult:"],
         }
-        intermediate_steps = []
-        sql_cmd = llm_chain.predict(**llm_inputs)
-        intermediate_steps.append(sql_cmd)
-        self.callback_manager.on_text(sql_cmd, color="green", verbose=self.verbose)
-        result = self.database.run(sql_cmd)
-        intermediate_steps.append(result)
-        self.callback_manager.on_text("\nSQLResult: ", verbose=self.verbose)
-        self.callback_manager.on_text(result, color="yellow", verbose=self.verbose)
-        # If return direct, we just set the final result equal to the sql query
-        if self.return_direct:
-            final_result = result
-        else:
-            self.callback_manager.on_text("\nAnswer:", verbose=self.verbose)
-            input_text += f"{sql_cmd}\nSQLResult: {result}\nAnswer:"
-            llm_inputs["input"] = input_text
-            final_result = llm_chain.predict(**llm_inputs)
-            self.callback_manager.on_text(
-                final_result, color="green", verbose=self.verbose
-            )
-        chain_result: Dict[str, Any] = {self.output_key: final_result}
-        if self.return_intermediate_steps:
-            chain_result["intermediate_steps"] = intermediate_steps
-        return chain_result
+        intermediate_steps: List = []
+        try:
+            intermediate_steps.append(llm_inputs)
+            sql_cmd = llm_chain.predict(**llm_inputs)
+            intermediate_steps.append(sql_cmd)
+            if not self.use_query_checker:
+                self.callback_manager.on_text(
+                    sql_cmd, color="green", verbose=self.verbose
+                )
+                result = self.database.run(sql_cmd)
+            else:
+                query_checker_prompt = self.query_checker_prompt or PromptTemplate(
+                    template=QUERY_CHECKER, input_variables=["query", "dialect"]
+                )
+                query_checker_chain = LLMChain(
```

Comment:
@question: is it known that the query checker will not return the SQL in double quotes?

Reply:
The prompt for the query checker is here: https://github.com/hwchase17/langchain/blob/master/langchain/tools/sql_database/prompt.py. It does ask the model to properly quote identifiers, but does not say anything about SQL in double quotes... so I guess the answer to your question is ... it might.